### PR TITLE
Add utility functions to map from Wayland types to MirAL types

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

**Comment by wmww** [comment]
bors try

File: tests/miral/org_kde_kwin_server_decoration.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/* Generated by wayland-scanner 1.16.0 */
```
Comment (wmww) [inline tests/miral/org_kde_kwin_server_decoration.c]: I get this is a test, but even so why is there a generated C file?

Comment (AlanGriffiths) [reply tests/miral/org_kde_kwin_server_decoration.c]: Simpler than ensuring `wayland-scanner` is available and generating it every time. There's a `.h` too.

Comment (wmww) [reply tests/miral/org_kde_kwin_server_decoration.c]: Sounds good

**Pull request merged**